Code/Tools/AssetProcessor/native/utilities/GUIApplicationManager.cpp

```diff
@@ -71,6 +72,25 @@ namespace
 
 }
 
+ErrorCollector::~ErrorCollector()
+{
+    if (!m_errorMessages.isEmpty())
+    {
+        MessageWindow messageWindow(m_parent);
+        messageWindow.SetHeaderText("The following errors occurred during startup:");
+        messageWindow.SetMessageText(m_errorMessages);
+        messageWindow.SetTitleText("Startup Errors");
+        messageWindow.exec();
+    }
+}
+
+void ErrorCollector::AddError(AZStd::string message)
+{
+    QString qMessage(message.c_str());
+    qMessage = qMessage.trimmed();
+
+    m_errorMessages << qMessage;
```

Comment:
should this be made thread safe? Or do these messages only ever come in on the same thread? When I fixed up the Editor's startup error handling a few years back, that was a big part of it, making it thread safe.

Reply:
The only entry point into this is TraceMessageBus OnError in GUIApplicationManager, and that ebus has a recursive mutex on it, so it's safe for now.